nav2_theta_star_planner/src/theta_star_planner.cpp

```diff
@@ -93,8 +93,7 @@ nav_msgs::msg::Path ThetaStarPlanner::createPlan(
   planner_->costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, mx_goal, my_goal);
   if (mx_start == mx_goal && my_start == my_goal) {
```

Comment:
Isn't this start=goal a planner error type?

I think you should carefully review the smac/navfn/theta* planners and make sure that for the 8 types of errors you have enumerated, each (where relevant) is thrown in the planner algorithms

Reply:
So, 

the angle could be different that is just check the x y position. 

Those eight exceptions don't necessary need to be throw by the all the planners. I went through and substituted out warning for throwing exceptions. If the planners doesn't care about a particular error I ignored it. (thought it was outside the scope :)). 

If we want to keep consistent across planners generic checks should be done up front in the planner server rather than the plugin.

The planner server would throw:
1. goal/start occupied 
2. start/goal outside map 
3. start==goal

The planners would throw: 
1. NoValidPathCouldBeFound
2. PlannerTimedOut.

Reply:
Out of scope but something I was just thinking about. 

If a goal is blocked it be quite useful to come up with some way of adjusting the goal to a valid cell.

Reply:
> the angle could be different that is just checking the x y position.

That's only important for feasible planners (e.g. Hybrid-A* and State Lattice). The angle otherwise isn't considered with the rest since they're holonomic planners. That's kind of the main reason those checks can't happen in the planner server itself - it has no knowledge about which planners care about orientation and which don't. 

I agree that the 8 exceptions do not need to be thrown by all planners. I guess what I'm asking for is some consistency over all of the algorithms that they throw the "core" ones (goal/start occupied; off map; or equal) since we're now actually changing the behavior of the server to catch and send these up to the autonomy application. If a user has a BT that relies on these codes for recovery actions, then it stands to reason that if they change to any of the "supported" algorithms, they should be interchangable. 

The only way to get that is if each of the plugins then throws the all of the exceptions that are relevant to them, even if right now they didn't previously log a complaint. For example, the Theta* planner doesn't have a time limit, so it wouldn't be sensible for it to throw a `TIMEDOUT` error. But all of the planners can check if `start=goal` and have the identical behavior. For the feasible planners, that check should include orientation, else, it shouldn't. 

> If a goal is blocked it would quite useful to come up with some way of adjusting the goal to a valid cell.

NavFn and Smac2D do this. Navfn will backtrack the path to the goal until it is outside of the set `tolerance` for something valid. Smac2D has the `iterations_on_approach` to check for the number of iterations within tolerance, once exceeded, it returns the best result. We don't have this for the feasible planners (yet). I'm not sure why we don't have it for Theta*.